app/src/main/java/com/snowtam/io/ui/main/MainFragment.java

```diff
@@ -37,31 +47,59 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
                              @Nullable Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.main_fragment, container, false);
 
-        etSearch = view.findViewById(R.id.editText_Search);
         motionLayout = (MotionLayout) view.findViewById(R.id.main);
+        buttonAddSearch = (Button) view.findViewById(R.id.button_add_et_search);
+        recyclerView = (RecyclerView) view.findViewById(R.id.recyclerView_editText_search);
+        recyclerView.setHasFixedSize(true);
+        LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());
+        recyclerView.setLayoutManager(layoutManager);
+
+        // we are null to display the fist EditText search
+        list_et_search.add(null);
 
-        etSearch.setOnTouchListener(getOnTouch(view));
 
+        mAdapter = new DataAdapterSearch(getActivity(), list_et_search);
+        recyclerView.setAdapter(mAdapter);
+
+        recyclerView.addOnItemTouchListener(getTouchListener());
+
+        buttonAddSearch.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                list_et_search.add(null);
+                mAdapter.notifyDataSetChanged();
+            }
+        });
 
         return view;
     }
 
+    private RecyclerView.OnItemTouchListener getTouchListener() {
```

Comment:
Why this?

Reply:
This method used to apply a listener to the recycleviewSearch, the listener will trigger the animation whenever the user touches the recycleview

Reply:
All right!